mne_bids/utils.py

```diff
@@ -29,6 +30,143 @@
+    run : int | None
+        The run number for this item. Corresponds to "run".
+    processing : str | None
+        The processing label for this item. Corresponds to "proc".
+    recording : str | None
+        The recording name for this item. Corresponds to "recording".
+    space : str | None
+        The coordinate space for an anatomical file. Corresponds to "space".
+    prefix : str | None
+        The prefix for the filename to be created. E.g., a path to the folder
+        in which you wish to create a file with this name.
+    suffix : str | None
+        The suffix for the filename to be created. E.g., 'audio.wav'.
+
+    Returns
+    -------
+    filename : str
+        The BIDS filename you wish to create.
+
+    Examples
+    --------
+    >>> bids_basename = make_bids_basename(subject='test', session='two', task='mytask', suffix='data.csv')  # noqa: E501
+    >>> print(bids_basename)
+    sub-test_ses-two_task-mytask_data.csv
+    >>> print(bids_basename.__repr__())
+    BIDSPath (sub-test_ses-two_task-mytask_data.csv)
+    """
+
+    def __init__(self, subject=None, session=None,
+                 task=None, acquisition=None, run=None, processing=None,
+                 recording=None, space=None, prefix=None, suffix=None,
+                 **kwargs):
+        # allows shortened aliases for some entity values
+        if subject is None:
+            subject = kwargs.get('sub', None)
+        if session is None:
+            session = kwargs.get('ses', None)
+        if acquisition is None:
+            acquisition = kwargs.get('acq', None)
+        if processing is None:
+            processing = kwargs.get('proc', None)
+        if recording is None:
+            recording = kwargs.get('rec', None)
+
+        self.sub = subject
+        self.ses = session
+        self.task = task
+        self.acq = acquisition
+        self.run = run
+        self.proc = processing
+        self.rec = recording
+        self.space = space
+        self.prefix = prefix
+        self.suffix = suffix
+
+        # run string representation to check validity of arguments
+        str(self)
+
+    @property
+    def entities(self):
+        """Return dictionary of the BIDS entities."""
+        keys = ('sub', 'ses', 'task', 'acq',
+                'proc', 'acq', 'run', 'rec',
+                'space', 'suffix', 'prefix')
+        entities = OrderedDict()
+
+        for key in keys:
+            value = getattr(self, key, None)
+            if value is not None:
+                entities[key] = value
+        return entities
+
+    def __setitem__(self, key, value):
+        """Set item as a dictionary, and perform validation checks."""
+        if key not in ('sub', 'ses', 'task', 'acq',
+                       'proc', 'acq', 'run', 'rec',
+                       'space', 'suffix', 'prefix'):
+            raise ValueError('Key must be one of blah, got %s' % key)
+        setattr(self, key, value)
```

Comment:
do you use this in the code anywhere? I would remove any magic methods that are not used in the code anywhere and make the class minimal

Reply:
Sorry this was leftover from when we were using it as a dict.